### DISCUSSION TITLE
Input-Leap, Fedora 40, and Wayland

### DISCUSSION THREAD

**Discussion by pentastich** [input-leap/input-leap#1903] [Q&A]
My apologies for asking yet another Wayland question!

I just installed Fedora 40, and both input-leap (2.4.0^20231022gitc5bb9dca-3.fc40) and libportal (0.7.1-3.fc40) are included. Fedora 40 comes with Gnome 46.1. Unfortunately, if I try to start up input-leap under wayland, I get an endless series of errors like the following:

> [2024-04-29T15:49:03] INFO: starting server
[2024-04-29T15:49:03] INFO: config file: /tmp/InputLeap.jkLTns
[2024-04-29T15:49:03] INFO: log level: INFO
[2024-04-29T15:49:03] FATAL: An error occurred: Missing libportal InputCapture portal support
[2024-04-29T15:49:03] ERROR: process exited with error code: 1
[2024-04-29T15:49:03] INFO: detected process not running, auto restarting

Furthermore, hitting "Stop" on the "InputLeap" screen causes my screen to go blank and the gnome-session to crash.

I know you're not the Fedora team, but do you have any idea what's going on, and what my best path forward is?

I'd been using input-leap under X11 on Fedora 39, and going back to an X11 desktop is an option. I was hoping to be able to cut over to Wayland with Fedora 40, though.

[Note: the session crash does NOT happen if I start "input-leap" from the command line. It only happens if I start "input-leap" through the Gnome GUI]

**Comment by krypticbit** [comment]
I have also had this issue. I managed to fix it by installing the input-leap-ei-enabled version from here: https://copr.fedorainfracloud.org/coprs/ofourdan/input-leap-ei-enabled/

```
sudo dnf remove input-leap
sudo dnf copr enable ofourdan/input-leap-ei-enabled
sudo dnf install input-leap --repo copr:copr.fedorainfracloud.org:ofourdan:input-leap-ei-enabled
sudo dnf update
```

Note that the final update command is important, my system didn't start working until I updated it. I believe it replaces libportal with patched versions from the copr repo.

**Reply by pentastich** [reply]
Thanks. That worked to get it up and running. Unfortunately, that version of input-leap (all versions? who knows?) still seems to have the problem where it messes up modifier keys (CTL, etc.).

**Reply by shymega** [reply]
Known bug.

**Comment by pentastich** [comment]
Thanks. I tried it, and it runs that way. Unfortunately that version, at least, still has the modifier key problems so, for example, "Control" is sticky on the client machine. Some day!

**Comment by Gorlinux** [comment]
Got the same issue on Nobara (Fedora FC40 + KDE + Wayland).
Saw that most available libportal versions are 0.7.1-4.fc40 execept one:

```
root@x9b:~# dnf install  libportal* 2>/dev/null | grep libportal
 libportal-debuginfo      x86_64 0.7.1-4.fc40   nobara-baseos-40          174 k
 libportal-debugsource    x86_64 0.7.1-4.fc40   nobara-baseos-40           60 k
 libportal-devel-doc      noarch 0.7.1-4.fc40   nobara-baseos-40           96 k
 libportal-gtk3           i686   0.7.1-4.fc40   nobara-baseos-multilib-40  17 k
 libportal-gtk3-debuginfo x86_64 0.7.1-4.fc40   nobara-baseos-40           23 k
 libportal-gtk3-devel     i686   0.7.1-4.fc40   nobara-baseos-multilib-40  11 k
 libportal-gtk4           i686   0.7.1-4.fc40   nobara-baseos-multilib-40  17 k
 libportal-gtk4-debuginfo x86_64 0.7.1-4.fc40   nobara-baseos-40           23 k
 libportal-gtk4-devel     i686   0.7.1-4.fc40   nobara-baseos-multilib-40  11 k
 libportal-qt5            i686   0.7.1-4.fc40   nobara-baseos-multilib-40  27 k
 libportal-qt5-debuginfo  x86_64 0.7.1-4.fc40   nobara-baseos-40          201 k
 libportal-qt5-devel      i686   0.7.1-4.fc40   nobara-baseos-multilib-40  11 k
 libportal-qt6-devel      x86_64 0.8.1-1.fc40   updates                    10 k
 libportal                i686   0.7.1-4.fc40   nobara-baseos-multilib-40  77 k
 libportal-devel          i686   0.7.1-4.fc40   nobara-baseos-multilib-40  43 k
 libportal                x86_64 0.7.1-4.fc40   nobara-baseos-40           75 k
 libportal-devel          x86_64 0.7.1-4.fc40   nobara-baseos-40           43 k
 libportal-gtk3           x86_64 0.7.1-4.fc40   nobara-baseos-40           18 k
 libportal-gtk3-devel     x86_64 0.7.1-4.fc40   nobara-baseos-40           11 k
 libportal-gtk4           x86_64 0.7.1-4.fc40   nobara-baseos-40           18 k
 libportal-gtk4-devel     x86_64 0.7.1-4.fc40   nobara-baseos-40           11 k
 libportal-qt5            x86_64 0.7.1-4.fc40   nobara-baseos-40           27 k

```

Solution: 
sudo dnf install libportal-qt6

Issue is gone now, even with stock input-leap.
root@x9b:~# dnf info input-leap | grep Version
Version      : 2.4.0^20240928gitb7bceeb2
